arm/Microsoft.Compute/virtualMachines/.bicep/nested_networkInterface.bicep

```
@@ -16,108 +19,83 @@ param diagnosticEventHubName string
+    location: location
     lock: lock
-    roleAssignments: contains(ipConfiguration.pipconfiguration, 'roleAssignments') ? (!empty(ipConfiguration.pipconfiguration.roleAssignments) ? ipConfiguration.pipconfiguration.roleAssignments : []) : []
+    publicIPAddressVersion: contains(ipConfiguration, 'publicIPAddressVersion') ? ipConfiguration.publicIPAddressVersion : 'IPv4'
+    publicIPAllocationMethod: contains(ipConfiguration, 'publicIPAllocationMethod') ? ipConfiguration.publicIPAllocationMethod : 'Static'
+    publicIPPrefixResourceId: contains(ipConfiguration, 'publicIPPrefixResourceId') ? ipConfiguration.publicIPPrefixResourceId : ''
+    roleAssignments: contains(ipConfiguration, 'roleAssignments') ? ipConfiguration.roleAssignments : []
+    skuName: contains(ipConfiguration, 'skuName') ? ipConfiguration.skuName : 'Standard'
+    skuTier: contains(ipConfiguration, 'skuTier') ? ipConfiguration.skuTier : 'Regional'
     tags: tags
+    zones: contains(ipConfiguration, 'zones') ? ipConfiguration.zones : []
   }
 }]
 
-resource networkInterface 'Microsoft.Network/networkInterfaces@2021-05-01' = {
-  name: networkInterfaceName
-  location: location
-  tags: tags
-  properties: {
-    enableIPForwarding: enableIPForwarding
-    enableAcceleratedNetworking: enableAcceleratedNetworking
-    dnsSettings: !empty(dnsServers) ? {
-      dnsServers: dnsServers
-    } : null
-    networkSecurityGroup: !empty(networkSecurityGroupId) ? {
-      id: networkSecurityGroupId
-    } : null
-    ipConfigurations: [for (ipConfiguration, index) in ipConfigurationArray: {
+module networkInterface '../../../Microsoft.Network/networkInterfaces/deploy.bicep' = {
+  name: '${deployment().name}-NetworkInterface'
+  params: {
+    name: networkInterfaceName
+    ipConfigurations: [for (ipConfiguration, index) in ipConfigurations: {
       name: !empty(ipConfiguration.name) ? ipConfiguration.name : null
-      properties: {
-        primary: ((index == 0) ? true : false)
-        privateIPAllocationMethod: contains(ipConfiguration, 'privateIPAllocationMethod') ? (!empty(ipConfiguration.privateIPAllocationMethod) ? ipConfiguration.privateIPAllocationMethod : null) : null
-        privateIPAddress: contains(ipConfiguration, 'vmIPAddress') ? (!empty(ipConfiguration.vmIPAddress) ? ipConfiguration.vmIPAddress : null) : null
-        publicIPAddress: contains(ipConfiguration, 'pipconfiguration') ? json('{"id":"${resourceId('Microsoft.Network/publicIPAddresses', '${virtualMachineName}${ipConfiguration.pipconfiguration.publicIpNameSuffix}')}"}') : null
-        subnet: {
-          id: ipConfiguration.subnetId
-        }
-        loadBalancerBackendAddressPools: contains(ipConfiguration, 'loadBalancerBackendAddressPools') ? ipConfiguration.loadBalancerBackendAddressPools : null
-        applicationSecurityGroups: contains(ipConfiguration, 'applicationSecurityGroups') ? ipConfiguration.applicationSecurityGroups : null
-      }
+      primary: index == 0 ? true : false
```

Simplified, if index is 0, the statement is true, else it is false. No need for a ternary operator here.

```suggestion
      primary: index == 0
```